src/riscv/interpreter.rs

```diff
@@ -0,0 +1,75 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use bytes::Bytes;
+use ckb_vm::machine::SupportMachine;
+
+use crate::evm::DataProvider;
+use crate::riscv;
+use crate::{Context, InterpreterParams, InterpreterResult};
+
+pub struct Interpreter {
+    pub context: Context,
+    pub iparams: InterpreterParams,
+    pub data_provider: Rc<RefCell<DataProvider>>,
```

Comment:
Why are `Rc` in some places and `Arc` in others?

I think you should take this opportunity to change them all to `Rc` because the `VM` is not thread-safe.

Reply:
Some low level data structures, like `block_provider` or `state_provider`, It may be used concurrently by RPC module, or something else. The interpreter is single core running, but the state or database is not.

```
pub struct DataProvider<B> {
    pub block_provider: Arc<BlockDataProvider>,
    pub state_provider: Arc<RefCell<State<B>>>,
    pub store: Arc<RefCell<Store>>,
}
```

I will consider use `Rc` instead of `Arc` careful, maybe in next PR.